rotkehlchen/tests/api/test_evm_transactions.py

```diff
@@ -89,3 +129,112 @@ def test_query_transactions(rotkehlchen_api_server: 'APIServer'):
+            chain_id=chain_id,
+            address=ADDY,
+            transaction_should_exists=False,
+        )
+
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'evmtransactionshashresource',
+        ), json={
+            'async_query': is_async_query,
+            'evm_chain': chain_id.to_name(),
+            'tx_hash': tx_hash,
+            'associated_address': ADDY,
+        },
+    )
+    if is_async_query is True:
+        task_id = assert_ok_async_response(response)
+        result = wait_for_async_task_with_result(rotkehlchen_api_server, task_id)
+        assert result is True
+    else:
+        assert_simple_ok_response(response)
+
+    # now see that the transaction is present in the db
+    # and the transaction was decoded properly
+    with database.conn.read_ctx() as cursor:
+        _assert_evm_transaction_status(
+            cursor=cursor,
+            tx_hash=tx_hash,
+            chain_id=chain_id,
+            address=ADDY,
+            transaction_should_exists=True,
+        )
+        cursor.execute('SELECT * FROM history_events WHERE event_identifier=?', (hexstring_to_bytes(tx_hash),))  # noqa: E501
+        events = [HistoryBaseEntry.deserialize_from_db(entry) for entry in cursor]
+        assert expected_decoded_events == events
+
+    # check for errors
+    # use an unsupported evm chain and see that it fails
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'evmtransactionshashresource',
+        ), json={
+            'async_query': is_async_query,
+            'evm_chain': ChainID.ARBITRUM.to_name(),
+            'tx_hash': tx_hash,
+            'associated_address': ADDY,
+        },
+    )
+    assert_error_response(response, 'Given chain_id arbitrum is not one of ethereum,optimism as needed by the endpoint')  # noqa: E501
+
+    # add an already existing transaction
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'evmtransactionshashresource',
+        ), json={
+            'async_query': is_async_query,
+            'evm_chain': chain_id.to_name(),
+            'tx_hash': tx_hash,
+            'associated_address': ADDY,
+        },
+    )
+    assert_error_response(response, 'transaction hash already present in the database')
+
+    # use an associated address that is not tracked by rotki
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'evmtransactionshashresource',
+        ), json={
+            'async_query': is_async_query,
+            'evm_chain': chain_id.to_name(),
+            'tx_hash': make_evm_tx_hash(make_random_bytes(32)).hex(),  # pylint: disable=no-member
+            'associated_address': make_evm_address(),
+        },
+    )
+    assert_error_response(response, 'address provided is not tracked by rotki for the specified evm chain')  # noqa: E501
```

Comment:
Another error that you need to check for (and have a proper error for) is adding a transaction hash that does not exist (for the chain you provide)